src/jllvm/vm/StringInterner.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <llvm/Support/raw_ostream.h>
+
+#include <jllvm/object/ClassLoader.hpp>
+#include <jllvm/support/Encoding.hpp>
+
+#include <map>
+
+namespace jllvm
+{
+class StringInterner
+{
+    static constexpr auto stringDescriptor = "Ljava/lang/String;";
+    static constexpr auto byteArrayDescriptor = "[B";
+
+    std::map<llvm::StringRef, String*> m_literalToStringMap;
+    llvm::BumpPtrAllocator m_stringAllocator;
+    llvm::BumpPtrAllocator m_valueAllocator;
```

Comment:
Is there any benefit to having these be separate allocators or would just one large `llvm::BumpPtrAllocator` suffice?